week-02/assessment/gomoku/src/main/java/learn/gomoku/GameController.java

```diff
@@ -0,0 +1,128 @@
+package learn.gomoku;
+
+import learn.gomoku.game.Gomoku;
+import learn.gomoku.game.Result;
+import learn.gomoku.game.Stone;
+import learn.gomoku.players.HumanPlayer;
+import learn.gomoku.players.Player;
+import learn.gomoku.players.RandomPlayer;
+
+import java.util.List;
+import java.util.Scanner;
+
+import static learn.gomoku.game.Gomoku.WIDTH;
+
+public class GameController {
+
+    private Scanner console = new Scanner(System.in);
+    Gomoku game;
+
+
+    public void run(){
+        setup();
+        play();
+        playAgain();
+    }
+
+
+    private void setup(){
+        System.out.println("Player 1 is: ");
+        Player player1 = getPlayer(1);  // gets one player
+        System.out.println("Player 2 is: ");
+        Player player2 = getPlayer(2);
+        game = new Gomoku(player1, player2);
+    }
+
+    private Player getPlayer(int playerNumber){
+
+            playerNumber = readInt("\n1. Human\n2. Random Player\nSelect [1-2]: ", 1, 2);
```

Comment:
MAJOR: playerNumber is an INPUT to this function that you're just immediately overwriting with the user choice about player TYPE (which is not meant to be the same thing).  use player number to print out the messages you're currently printing in setup() and then create a NEW variable to hold the user's choice of player type.